src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -279,9 +279,42 @@ public AzureAppConfigurationOptions Connect(Uri endpoint, TokenCredential creden
                 throw new ArgumentNullException(nameof(credential));
             }
 
+            return Connect(new List<Uri>() { endpoint }, credential);
+        }
+
+        /// <summary>
+        /// Connect the provider to Azure App Configuration and their replicas using list of endpoints and token credential.
+        /// </summary>
+        /// <param name="endpoints">The list of endpoints of the Azure App Configuration and its replicas to connect to.</param>
+        /// <param name="credential">Token credential to use to connect.</param>
+        public AzureAppConfigurationOptions Connect(IEnumerable<Uri> endpoints, TokenCredential credential)
+        {
+            if (endpoints == null || endpoints.Count() == 0)
+            {
+                throw new ArgumentNullException(nameof(endpoints));
+            }
+
+            Credential = credential ?? throw new ArgumentNullException(nameof(credential));
+
+            if (endpoints.Count() > 1)
+            {
+                string firstEndpoint = endpoints.First().Host;
+
+                // If the first endpoint is a replica or the store name has a '-', the common prefix would be string before first '-'
+                // else if there is no '-' in it, then it is not a replica and the common prefix would be a substring before first '.'
+                string commonPrefix = firstEndpoint.Substring(0, firstEndpoint.IndexOf('-') > 0 ? firstEndpoint.IndexOf('-') : firstEndpoint.IndexOf('.'));
+                string commonDomain = firstEndpoint.Substring(firstEndpoint.IndexOf('.'));
```

Comment:
> commonDomain

That's most likely going to break with regional replication.

Reply:
This comment is not applicable anymore since we removed the validations.